core/server/lib/common/i18n.js

```diff
@@ -115,22 +151,103 @@ I18n = {
+                // polyfill and replace the constructors with need with the polyfill's.
+                IntlPolyfill = require('intl');
+                Intl.NumberFormat   = IntlPolyfill.NumberFormat;
+                Intl.DateTimeFormat = IntlPolyfill.DateTimeFormat;
+            }
+        } else {
+            // No `Intl`, so use and load the polyfill.
+            global.Intl = require('intl');
+        }
+    },
+
+    /**
+     * Setup i18n support for themes:
+     *  - Load proper language file in to memory
+     *  - Polyfill node.js if it does not have Intl support or support for a particular locale
+     */
+    loadThemeTranslations: function loadThemeTranslations() {
+        // This function is called during theme initialization, and when switching language or theme.
+
+        var hasBuiltInLocaleData, IntlPolyfill;
+
+        currentLocale = I18n.locale();
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading file for current locale and active theme and keeping its content in memory
+        if (activeTheme) {
+            // Reading translation file for theme .hbs templates.
+            // Compatibility with both old themes and i18n-capable themes.
+            // Preventing missing files.
+            try {
+                themeStrings = fs.readFileSync(path.join(__dirname, '..', '..', '..', '..', 'content', 'themes', activeTheme, 'locales', currentLocale + '.json'));
+            } catch (err) {
+                themeStrings = undefined;
+                if (err.code === 'ENOENT') {
+                    logging.warn('Theme\'s file locales/' + currentLocale + '.json not found.');
+                } else {
+                    throw err;
+                }
+            }
+            if (themeStrings === undefined && currentLocale !== 'en') {
+                logging.warn('Falling back to locales/en.json.');
+                try {
+                    themeStrings = fs.readFileSync(path.join(__dirname, '..', '..', '..', '..', 'content', 'themes', activeTheme, 'locales', 'en.json'));
+                } catch (err) {
+                    themeStrings = undefined;
+                    if (err.code === 'ENOENT') {
+                        logging.warn('Theme\'s file locales/en.json not found.');
+                    } else {
+                        throw err;
+                    }
+                }
+            }
+            if (themeStrings !== undefined) {
+                // if translation file is not valid, you will see an error
+                try {
+                    themeStrings = JSON.parse(themeStrings);
+                } catch (err) {
+                    themeStrings = undefined;
+                    throw err;
+                }
+            }
+        }
+
+        if (themeStrings === undefined) {
+            // even if empty, themeStrings must be an object for jp.value
+            themeStrings = {};
+        }
+
+        if (global.Intl) {
```

Comment:
Could you pls outsource this to a function `_private.initialiseIntl` - it's now occurring twice in the file 👍 Makes the file more readable.

Reply:
Done, the two instances of _Intl_ polyfill consolidated into one function `_private.initializeIntl()`. Implemented similarly to all Ghost's `_private.` functions.